pkg/operation/botanist/secrets.go

```diff
@@ -244,6 +251,55 @@ func (b *Botanist) rotateKubeconfigSecrets(ctx context.Context, gardenerResource
 	return b.K8sGardenClient.Client().Patch(ctx, b.Shoot.Info, client.MergeFrom(oldObj))
 }
 
+func (b *Botanist) rotateSSHKeypairSecrets(ctx context.Context, gardenerResourceDataList *gardencorev1alpha1helper.GardenerResourceDataList) error {
+	var err error
+
+	// fetch the current SSH keypair
+	currentSecret := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: v1beta1constants.SecretNameSSHKeyPair, Namespace: b.Shoot.SeedNamespace}}
+
+	if err = b.K8sSeedClient.Client().Get(ctx, client.ObjectKeyFromObject(currentSecret), currentSecret); client.IgnoreNotFound(err) != nil {
```

Comment:
If the secret is not found, can the following code lead to panics?

Reply:
Nope, `oldPrivateKey := currentSecret.Data[secrets.DataKeyRSAPrivateKey]` won't panic and the following code doesn't either. Just tested by deleting all SSH keypair secrets in garden/seed and the gardenlet happily reconciled the state.

Reply:
👍🏻 thanks!